core/FrontController.php

```diff
@@ -504,6 +504,10 @@ protected function handleMaintenanceMode()
             return;
         }
 
+        if ($this->isRequestToCoreApiHealthCheck() === true) {
```

Comment:
Had a closer look what doing a return here in maintenance mode actually causes:
The whole platform will be initialized, which will also trigger a database connection to be established and all plugins to be loaded.
But actually the maintenance mode option in config is commented with
```
when set to 1, all requests to Matomo will return a maintenance message without connecting to the DB
```
so a database connection should be avoided. I doubt that we can easily adjust the code to avoid a database connection while still loading all required stuff to handle API requests correctly. Even loading a certain plugin might already require a database connection.
Overall we currently might not be able to support the maintenance stuff in the healthcheck api they way it is currently implemented. We could though have a complete separate handling of that API method in the front controller directly. So it doesn't need to initialize the whole platform, but only the parts required to handle the API response formats and stuff like that.

Reply:
Nice spot!

I don't know if it's really important that the maintenance mode even passes through to our nice health check responses.

How about doing this:

- Our health check API supports only JSON response
- If site in maintenance mode and request made with header `Accepts: application/json` or `&format=JSON` then respond early with empty body and `503` status.
- Update the `DashboardAvailable` with a different implementation, might even just return true for now.

Reply:
Additionally I wanted to check 
`when set to 1, all requests to Matomo will return a maintenance message without connecting to the DB`

I can see that locally it still attempts to connect in maintenance mode, I purposively put the wrong credentials in and tailed the database connection logs.

<img width="1041" alt="image" src="https://github.com/matomo-org/matomo/assets/1169490/d9b5446f-790e-4ac1-9855-b356e68f63ec">

Reply:
However I've gone ahead and moved this into core, it will be hit before the maintenance screen, however this means our check if the dashboard plugin has been loaded is not possible, so I at least just do a check that its configured to be loaded.